specification.md

```diff
@@ -1925,11 +1925,10 @@ definition "id" must be a constant value.
 
 | Parameter | Description | Type | Required |
 | --- | --- |  --- | --- |
-| id | Workflow unique identifier | string | yes (if `key` is not defined) |
-| key | Domain-specific workflow identifier | string | yes (if `id` is not defined) |
-| name | Workflow name | string | no |
+| name | Workflow definition unique identifier (which must be human readable) | string | yes |
+| version | Workflow version. MUST respect the [semantic versioning](https://semver.org/) format. If not provided, latest is assumed | string | no |
 | description | Workflow description | string | no |
-| version | Workflow version. MUST respect the [semantic versioning](https://semver.org/) format | string | no |
+| key | Expression that will be used to generate a domain-specific workflow instance identifier  | string | no |
```

Comment:
I left key name as it is. In an ideal world, probably the name should be something like instance-key-generator, but to generate another backward incompatibility because of that seems not pratical to me, so I vote for the name "key" to stand